### DISCUSSION TITLE
Need a way to upgrade codeql cli automatically

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#10379] [General]
How can I upgrade codeql from cli?
I mean, I had to download the binary packages on github again, is there anything similar to `nuclei -update`?

**Comment by MathiasVP** [comment]
Hi @CaledoniaProject,

There is no built-in way to update the CLI directly, no.

If you're on a Mac (and using Homebrew), you can update it through that: https://formulae.brew.sh/cask/codeql. But I'm not aware of any other package managers that has a CodeQL entry.

**Reply by CaledoniaProject** [reply]
Is this feature planned?

**Reply by MathiasVP** [reply]
Not as far as I know. I'll bring it up to the CLI team. What do you need this feature for?

**Reply by CaledoniaProject** [reply]
I want to keep codeql and vscode-codeql-starter up-to-date, I would like to implement a cron job on my Linux server.

**Reply by CaledoniaProject** [reply]
Of course I can use the Github API and download the latest release, I just think the codeql cli should do it.

**Reply by MathiasVP** [reply]
Alternatively, you should take a look at https://github.com/github/gh-codeql. In particular, see the section of [Versions](https://github.com/github/gh-codeql#versions).

**Reply by CaledoniaProject** [reply]
Interesting, that should work for me.
